File: Analyzer/utils/getEvenPointsOnSphere.m
function [Vector] = getEvenPointsOnSphere(R,numberOfPoints,useGPU)
%GETEVENPOINTSONSPHERE Summary of this function goes here
%   Detailed explanation goes here

if useGPU
    goldenRatio = (1+5^0.5)/2;
    Vector = gpuArrayZeros(3,numberOfPoints);
    # numberOfPoints = gpuArray(numberOfPoints);
    R = gpuArray(R);
else
    goldenRatio = (1+5^0.5)/2;
    Vector = zeros(3,numberOfPoints);
end

for i = 0:numberOfPoints-1
    theta = 2*pi*i/goldenRatio;
    phi = acos(1-2*(i+0.5)/numberOfPoints);

    Vector(1,i+1) = R*cos(theta)*sin(phi);
    Vector(2,i+1) = R*sin(theta)*sin(phi);
    Vector(3,i+1) = R*cos(phi);
end

Vector = real(Vector);
# FIXME: OclArray: octave indexing must result in a contiguous memory range
if useGPU
    Vector = gather(Vector);
endif

end

File: analyzer/utils/get_even_points_on_sphere.py
import numpy as np

from constants.constants import golden_ratio


def get_even_points_on_sphere(radius: np.float64, number_of_points: int, use_gpu: bool = False) -> np.ndarray[np.float64]:
    vec = np.zeros((3, number_of_points))

    for i in range(number_of_points - 1):
        theta = 2 * np.pi * i / golden_ratio
        phi = np.acos(1 - 2 * (i + 0.5)/number_of_points)

        vec[0, i + 1] = radius * np.cos(theta) * np.sin(phi)
        vec[1, i + 1] = radius * np.sin(theta) * np.sin(phi)
        vec[2, i + 1] = radius * np.cos(phi)
    return np.real(vec)
